src/System.Net.Sockets/src/System/Net/Sockets/SafeSocketHandle.Unix.cs

```diff
@@ -187,226 +187,222 @@ internal void SetToDisconnected()
 
         internal static unsafe SafeSocketHandle CreateSocket(IntPtr fileDescriptor)
         {
-            return CreateSocket(InnerSafeCloseSocket.CreateSocket(fileDescriptor));
+            var res = new SafeSocketHandle();
+            res.SetHandle(fileDescriptor);
```

Comment:
Can we instead just pass the file descriptor to the SafeSocketHandle ctor?

Reply:
I was trying to ensure the `SafeSocketHandle` was allocated before assigning a handle, in case there wasn't enough memory. I'll change it.

Reply:
Ah, in this case, that doesn't make sense. Because the fileDescriptor already exists.